### PR TITLE
Feature/refactoring

### PR DESCRIPTION
**Pull request by PhilippRadler**

### CODE REVIEW

File: Classes/Controller/PersonController.php [hunk 1]
```diff
@@ -208,92 +241,35 @@ public function newAction(\Personmanager\PersonManager\Domain\Model\Person $newP
...
-        }
-        if (!filter_var(idn_to_ascii($newPerson->getEmail(),0,INTL_IDNA_VARIANT_UTS46), FILTER_VALIDATE_EMAIL)) {
-            $langhelp = LocalizationUtility::translate('error.email', $this->extKey);
-            $error .= "<p>$langhelp</p>";
-            $failed = 1;
-        }
+        $hash = $newPerson->getEmail() . time();
+        $newPerson->setToken(md5($hash));
+
         $oldMail = $this->personRepository->findOneByEmail($newPerson->getEmail());
```
Comment (parzerphilipp) [inline Classes/Controller/PersonController.php]: DB Request already made in EmailAgainValidator
maybe remove oldMail in Validator?

File: Classes/Controller/PersonController.php [hunk 2]
```diff
@@ -360,40 +321,41 @@ public function newLeaveAction()
      */
     public function leaveAction($mail = "")
     {
+        // Don't take the easy route - do it the hard way
         if ($mail == "") {
-            $mail = trim($this->request->getArguments()["mail"]);
+            $mail = trim($this->request->getArguments()["email"]);
```
Comment (parzerphilipp) [inline Classes/Controller/PersonController.php]: why email instead of mail?

**Review by parzerphilipp — CHANGES_REQUESTED** [review]
In Validators don't use time() - Error Code should be a static timestamp (current timestamp when you wrote the code)
Use [$this->translateErrorMessage](https://docs.typo3.org/m/typo3/book-extbasefluid/master/en-us/9-CrosscuttingConcerns/2-validating-domain-objects.html#validators-for-checking-of-invariants) instead of LocalizationUtility

File: Classes/Domain/Validator/TermsValidator.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+use TYPO3\CMS\Extbase\Validation\Validator\AbstractValidator;
+
+class TermsValidator extends AbstractValidator
+{
+   protected function isValid($value)
+   {
+       $terms = FormUtility::_GPmerged()['terms'];
+
+
+       $termVar = $GLOBALS['TSFE']->tmpl->setup["plugin."]["tx_personmanager."]["variables."]["terms"];
```
Comment (parzerphilipp) [inline Classes/Domain/Validator/TermsValidator.php]: could this be solved with Validator Options options={"terms": $this->settings} ?

File: Classes/Service/LogService.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     * @param CategoryRepository $logRepository
+     */
+    public function injectLogRepository(LogRepository $logRepository)
+    {
+        $this->logRepository = $logRepository;
+    }
+
+
+    public function insertLog($person = 0, $email = "", $fname = "", $lname = "", $action = "", $detail = "", $fehler = "", $success = 0)
```
Comment (parzerphilipp) [inline Classes/Service/LogService.php:51]: add PHPDoc Block

File: Classes/Service/PersonService.php [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        $pers->setActive(TRUE);
+        $this->personRepository->update($pers);
+        $this->persistenceManager->persistAll();
+
+        if ($sendInMail == 1) {
+            $this->mailService->doBuildActivateMail($pers, $msgKey, $to);
+        }
+
+        $langhelp = LocalizationUtility::translate($msgKey, $this->extKey);
+        $this->logService->insertLog($pers->getUid(), $pers->getEmail(), $pers->getFirstname(), $pers->getLastname(), $log, "$langhelp", "", 1);
```
Comment (parzerphilipp) [inline Classes/Service/PersonService.php]: remove "" for $langhelp

File: Resources/Private/Backend/Templates/Backend/BlNewImport.html [hunk 6]
```diff
@@ -1,42 +1,41 @@
...
-
-If you modify this template, do not forget to change the overwrite settings
-in /Configuration/ExtensionBuilder/settings.yaml:
-  Resources:
-    Private:
-      Templates:
-        New.html: keep
-
-Otherwise your changes will be overwritten the next time you save the extension in the extension builder
+<f:layout name="Default" /> This template displays a NEW form for the current domain object. If you modify this template, do not forget to change the overwrite settings in /Configuration/ExtensionBuilder/settings.yaml: Resources: Private: Templates: New.html: keep Otherwise your
```
Comment (parzerphilipp) [inline Resources/Private/Backend/Templates/Backend/BlNewImport.html]: remove description

File: Resources/Private/Backend/Templates/Backend/BlNewImport.html [hunk 7]
```diff
@@ -1,42 +1,41 @@
...
+changes will be overwritten the next time you save the extension in the extension builder
 
 <f:section name="content">
-  <h1><f:translate key="labels.importbl" /></h1>
+    <h1>
+        <f:translate key="labels.importbl" />
+    </h1>
 
-  <siwa:be.editLink parameters="edit[tx_personmanager_domain_model_blacklist][{vars.storagePid}]=new" returnAction="blNewImport" class="btn btn-success" title="{f:translate(key:'labels.newp')}"> &nbsp;<i class="fa fa-plus"></i> <f:translate key="labels.create" />&nbsp; </siwa:be.editLink>
+    <f:debug>{_all}</f:debug>
```
Comment (parzerphilipp) [inline Resources/Private/Backend/Templates/Backend/BlNewImport.html]: remove debug

File: Resources/Private/Backend/Templates/Backend/BlNewImport.html [hunk 8]
```diff
@@ -1,42 +1,41 @@
...
+    <siwa:be.editLink parameters="edit[tx_personmanager_domain_model_blacklist][{settings.storagePid}]=new" returnAction="blNewImport" class="btn btn-success" title="{f:translate(key:'labels.newp')}"> &nbsp;<i class="fa fa-plus"></i>
```
Comment (parzerphilipp) [inline Resources/Private/Backend/Templates/Backend/BlNewImport.html]: use be:uri.editRecord
xmlns:be="http://typo3.org/ns/TYPO3/CMS/Backend/ViewHelpers"

File: Resources/Private/Partials/Honeypot.html [hunk 9]
```diff
@@ -0,0 +1,11 @@
+<f:comment>
```
Comment (parzerphilipp) [inline Resources/Private/Partials/Honeypot.html]: remove comment

File: Resources/Private/Partials/Person/Fields/Salutation.html [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+		<f:translate key="tx_personmanager_domain_model_person.salutation" /> *
+	</label>
+    <br />
+    <f:form.select class="form-control" property="salutation" required="true">
+        <f:form.select.option value="">
+            <f:translate key="labels.choose" />
+        </f:form.select.option>
+        <f:if condition="{anr}==1">
+            <f:then>
+                <f:form.select.option value="1" selected="true">
```
Comment (parzerphilipp) [inline Resources/Private/Partials/Person/Fields/Salutation.html]: selected property can be condition {anr}==1

File: Resources/Private/Templates/Person/NewLeave.html [hunk 11]
```diff
@@ -17,7 +17,11 @@
 
 <f:form action="leave">
 
-    <f:form.textfield name="mail" /><br /><br />
+    <label for="email">
+      <f:translate key="tx_personmanager_domain_model_person.email" /> *
+    </label><br />
+    <f:form.textfield class="form-control" name="email" required="required" />
```
Comment (parzerphilipp) [inline Resources/Private/Templates/Person/NewLeave.html]: name is mail, not email

**Review by parzerphilipp — CHANGES_REQUESTED** [review]
fix TCA (EXT:lang deprecated) - modernize
TypoScript: rename files to .typoscript, remove CSS Default Style
clean ext_tables.sql

File: Classes/Domain/Validator/Person/EmailAgainValidator.php [hunk 12]
```diff
@@ -19,8 +18,9 @@ protected function isValid($person)
         $oldMail = $personRepository->findOneByEmail($person->getEmail());
         if ($oldMail != NULL) {
             if ($oldMail->isUnsubscribed() == 0) {
-               $langhelp = LocalizationUtility::translate('error.emailagain', 'person_manager');
-               $this->addError($langhelp, time());
+               $this->addError($this->translateErrorMessage('error.emailagain', 'person_manager'), 1620285838);
+            }else{
+                $personRepository->remove($oldMail);
```
Comment (parzerphilipp) [inline Classes/Domain/Validator/Person/EmailAgainValidator.php:23]: don't you have to persist this?

File: Resources/Private/Partials/Person/Fields/Salutation.html [hunk 13]
```diff
@@ -7,30 +7,12 @@
         <f:form.select.option value="">
             <f:translate key="labels.choose" />
         </f:form.select.option>
```
Comment (parzerphilipp) [inline Resources/Private/Partials/Person/Fields/Salutation.html:9]: isn't {anr} == 1 enough?

Comment (PhilippRadler) [reply Resources/Private/Partials/Person/Fields/Salutation.html:9]: probably

File: Classes/Domain/Validator/Person/EmailAgainValidator.php (hunk 12 above)
Comment (PhilippRadler) [reply Classes/Domain/Validator/Person/EmailAgainValidator.php:23]: tested it and deleted flag in db is set to 1 meaning you don't need the persistence manager

**Pull request merged**